util/include/Metrics.hpp

```diff
@@ -76,16 +78,19 @@ class Aggregator {
 template <class T>
 class BasicGauge {
  public:
+  typedef T type;
   explicit BasicGauge(const uint64_t val) : val_(val) {}
   BasicGauge(const BasicGauge& gauge) { val_ = (unsigned long)gauge.val_; }
   BasicGauge& operator=(const BasicGauge& gauge) {
     val_ = (unsigned long)gauge.val_;
     return *this;
   }
-  void Inc() { ++val_; }
-  void Dec() { --val_; }
+  // postfix
+  BasicGauge operator++(int) { return BasicGauge(val_++); }
+  // postfix
+  BasicGauge operator--(int) { return BasicGauge(val_--); }
   void Set(const uint64_t val) { val_ = val; }
```

Comment:
Nit: I know that this is an old code but I think it will be correct to change the argumant `const uint64_t val` to `const T& val`.

Reply:
Just missed it, will do it on another occasion - there are other things to refactor there